src/infrastructure/content/ContentCommunicator.java

```diff
@@ -1,24 +1,143 @@
+	 */
+	private String userName;
+	
+	/**
+	 * This variable will store the image of client in String format
+	 */
+	private String userImage;
+	
+	/**
+	 * This variable will store the server's ipAddress as a String
+	 */
+	private String serverIPAddress;
+	
+	/**
+	 * This variable will store the port number of the board which the client is accessing
+	 */
+	private int port = 0;
+	
+	/**
+	 * Creating a HashMap to store userName and userImage of all active clients of a particular board
+	 * Both userName and userImage are of String type
+	 */
+	private HashMap<String, String> mapImage = new HashMap<String, String>();
+	
+	/**
+	 * Creating a HashMap to store the identifier and handler of subscriber of this module
+	 * identifier is of type String while handler is of type IContentNotificationHandler
+	 */
+	private HashMap<String, IContentNotificationHandler> mapHandler = new HashMap<String, IContentNotificationHandler>();
+	
+	/**
+	 * Getting the communicator of networking module to send data over the network
+	 */
+	private ICommunicator communicator = CommunicatorFactory.getCommunicator(port);
+	
+	/**
+	 * This method provides userName to other classes within this package
+	 * @return userName - username of the client
+	 */
+	protected String getUserName() {
+		return userName;
+	}
+	
+	/**
+	 * This method provides HashMap mapImage to other classes within this package
+	 * @return mapImage - The HashMap which stores the userName and userImage
+	 */
+	protected HashMap<String, String> getMapImage() {
+		return mapImage;
+	}
+	
+	/**
+	 * This method allows other classes within this package to set the HashMap mapImage
+	 * @param mapImage - The HashMap which stores the userName and userImage of all clients
+	 */
+	protected void setMapImage(HashMap<String, String> mapImage) {
+		this.mapImage = mapImage;
+	}
+	
+	/**
+	 * This method provides HashMap mapHandler to other classes within this package
+	 * @return mapHandler - The HashMap which stores the identifier and handler of subscriber of this module
+	 */
+	protected HashMap<String, IContentNotificationHandler> getMapHandler() {
+		return mapHandler;
+	}
+	
+	/**
+	 * This method initializes a user by storing its username and userimage in a map as Strings
+	 * This method stores these values locally too and generates server's ipAddress by getting
+	 * 		the port value of the board from ServerPort class
+	 * Finally, some manipulation on JSON string is done and then sent to server via networking
+	 * @param userDetails - This is a JSON string which contains a JSON object with three fields, 
+	 * 		namely ipAddress, username and image.
+	 */
 	public void initialiseUser(String userDetails) {
-    	
-    }
+		JSONObject jsonObject = new JSONObject(userDetails);
+		serverIPAddress = jsonObject.getString("ipAddress");
+		userName = jsonObject.getString("username");
+		userImage = jsonObject.getString("image");
+		ServerPort serverPort = new ServerPort();
+		port = serverPort.getPort();

```

Comment:
What do you do if the serverPort object doesn't have the port value. You would need to wait for it right?

Reply:
It shouldn't be possible. The processing module sends port as soon as a new user joins a board. It will be done well before any methods of this class are called.
You should make sure that things happen this way by contacting with processing module.

Reply:
See, in the ideal case the process would be quick. What if there was a problem in the process and processing doesn't receive the port value? Proceeding without checks can be very risky.